### PR TITLE
Add variable for global flags to runlabel

### LINKED ISSUE #2399: Add variable for global flags to runlabel command
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

We need a variable in `podman runlabel` labels that expands to global Podman options, so we can pass through configuration (for example, storage config) to Podman commands being executed in labels.

**Steps to reproduce the issue:**

0. mkdir /tmp/new

1. podman --root=/tmp/new create --name test1 -t -i --label 'TEST=podman info'

2. podman --root=/tmp/new commit test1 testimg

3. podman --root /tmp/new container runlabel TEST testimg

**Describe the results you received:**

Podman uses the default storage root

**Describe the results you expected:**

Podman uses /tmp/new as a storage root

Ideal fix is something like $GLOBAL_OPTS we can throw in the label between `podman` and `info` to get the right options in there

### PR DESCRIPTION
**Pull request by QiWang19**
close #2399 
use $GLOBAL_OPTS to pass global flags to the runlabel command.
Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by QiWang19** [comment]
@mheon  will this satisfy the requirement in #2399 ? Then I need to look up the flags in GlobalFlags?

**Comment by mheon** [comment]
Initial impression is that this looks about right - just needs to pick up the rest of the local flags

**Comment by QiWang19** [comment]
do I need to add tests for this?

**Comment by mheon** [comment]
Tests are always appreciated!

**Comment by QiWang19** [comment]
Test added

File: cmd/podman/runlabel.go [hunk 1]
```diff
@@ -179,3 +180,66 @@ func runlabelCmd(c *cliconfig.RunlabelValues) error {
 
 	return utils.ExecCmdWithStdStreams(stdIn, stdOut, stdErr, env, cmd[0], cmd[1:]...)
 }
+
+func getGlobalOpts(c *cliconfig.RunlabelValues) string {
```
Comment (TomSweeneyRedHat) [inline cmd/podman/runlabel.go]: @mheon, should this live in a common or util file somewhere?

**Comment by TomSweeneyRedHat** [comment]
LGTM, just one question on a final resting place for one of the functions.

File: cmd/podman/runlabel.go (hunk 1 above)
Comment (QiWang19) [reply cmd/podman/runlabel.go]: I've moved it to package util

File: cmd/podman/shared/funcs.go [hunk 2]
```diff
@@ -60,6 +60,8 @@ func GenerateCommand(command, imageName, name string) ([]string, error) {
 	}
 	newCommand = append(newCommand, prog)
 
+	fmt.Println(strings.Join(cmd, ","))
```
Comment (mheon) [inline cmd/podman/shared/funcs.go]: Leftover debug?

File: pkg/util/utils.go [hunk 3]
```diff
@@ -324,3 +325,67 @@ func ParseInputTime(inputTime string) (time.Time, error) {
 	}
 	return time.Now().Add(-duration), nil
 }
+
+// GetGlobalOpts checks all global flags and generates the command string
+func GetGlobalOpts(c *cliconfig.RunlabelValues) string {
```
Comment (mheon) [inline pkg/util/utils.go]: I wonder if we can somehow pull the names of the global flags straight out of Cobra - @baude you know if there's a way to do this?

File: cmd/podman/shared/funcs.go (hunk 2 above)
Comment (QiWang19) [reply cmd/podman/shared/funcs.go]: yeah... deleted.

File: pkg/util/utils.go (hunk 3 above)
Comment (QiWang19) [reply pkg/util/utils.go]: I found FlagSet.VisitAll() for all flags, not just for global flags...

Comment (mheon) [reply pkg/util/utils.go]: Hm. This could be painful if we add more global flags, but no helping that, I suppose.

**Comment by mheon** [comment]
LGTM
@rhatdan @baude @vrothberg @TomSweeneyRedHat PTAL

File: pkg/util/utils.go [hunk 4]
```diff
@@ -252,3 +254,32 @@ func ParseInputTime(inputTime string) (time.Time, error) {
...
+	globalFlags := map[string]bool{
+		"cgroup-manager": true, "cni-config-dir": true, "conmon": true, "default-mounts-file": true,
```
Comment (rhatdan) [inline pkg/util/utils.go]: Separate PR, but could you move this to main_local.go
Add one line after each global flag gets added
```
	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.CGroupManager, "cgroup-manager", "", "Cgroup manager to use (cgroupfs or systemd, default systemd)")
        globalFlags["cgroup-manager"]=true
	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.CpuProfile, "cpu-profile", "", "Path for the cpu profiling results")
        globalFlags["cpu-profile"]=true
```

That way if a user adds a global flag, it is much more likely they will do the right thing.

**Pull request merged**